plugins/aea-cli-ipfs/aea_cli_ipfs/ipfs_utils.py

```diff
@@ -34,32 +34,62 @@ class IPFSDaemon:
     :raises Exception: if IPFS is not installed.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, offline: bool = False):
         """Initialise IPFS daemon."""
+
         # check we have ipfs
+        res = shutil.which("ipfs")
+        if res is None:
+            raise Exception("Please install IPFS first!")
+        process = subprocess.Popen(  # nosec
+            ["ipfs", "--version"],
+            stdout=subprocess.PIPE,
+            env=os.environ.copy(),
+        )
+        output, _ = process.communicate()
+        if b"0.6.0" not in output:
+            raise Exception(
+                "Please ensure you have version 0.6.0 of IPFS daemon installed."
+            )
         self.process = None  # type: Optional[subprocess.Popen]
+        self.offline = offline
 
     def is_started(self) -> bool:
         """Check daemon was started."""
         return bool(self.process)
 
     def start(self) -> None:
         """Run the ipfs daemon."""
+        cmd = ["ipfs", "daemon", "--offline"] if self.offline else ["ipfs", "daemon"]
         self.process = subprocess.Popen(  # nosec
-            ["ipfs", "daemon"], stdout=subprocess.PIPE, env=os.environ.copy(),
+            cmd,
+            stdout=subprocess.PIPE,
+            env=os.environ.copy(),
         )
+        for stdout_line in iter(self.process.stdout.readline, ""):
+            if b"Daemon is ready" in stdout_line:
+                break
```

Comment:
Instead of waiting for the daemon to be up, we can check the output from stdout to check if the daemon is up or not (It's really fast).